src/App.jsx

```diff
@@ -6,22 +6,30 @@ import './App.css'
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function App() {
 
-const useSemiPersistentState = () => {
-  const [todoList,setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
+  const [todoList,setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect (() => { 
+     const newPromise = new Promise((resolve, reject) => {
```

Comment:
unused variable newPromise. 
Compare 
<img width="496" alt="image" src="https://github.com/user-attachments/assets/bd555b81-1c02-4ec2-b895-cf249f8ea9ad">
and 
<img width="681" alt="image" src="https://github.com/user-attachments/assets/a6b1e3af-9a8c-4a80-94eb-41ede220c837">
(first example - without variable, second - with variable. [link on article](https://javascript.info/promise-chaining)